### PR TITLE
[feat][Android] Add ability to show subtitles and style them

### PR DESCRIPTION
**Pull request by fxdemolisher**
- Android: Added utility for easier extraction of info from ReadableMap instances
- Android: Split media info builder to its own class
- Android: Added ability to pass `textTrackStyle`, which allows caption styling
- Android: Added ability to enable/disabled captions
- RN: Modified example app to show the rest of the buttons on the android toolbar

Subtitle functionality provided by @ethanperez
Text track styling provided by @fxdemolisher

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -47,6 +68,9 @@ export default {
     contentType?: string,
     streamDuration?: number,
     playPosition?: number,
+    isLive?: Boolean,
```
Comment (nbennink) [inline index.js]: This refers to the class `Boolean`. `boolean` is the right type to use here.

I was wondering how I can add a MediaTrack to the item that I'm casting. Do I need to pass the url to the VTT files in customData?

Comment (fxdemolisher) [reply index.js]: Correcting to `boolean`, thanks.

As for attaching VTT files, we would need to alter `GoogleCastModule#castMedia` to accept extra options. Out of those we would need to build a MediaTrack instance to add to the MediaInfo object as per https://developers.google.com/cast/docs/android_sender/media_tracks

My change simply enables a specific (or default's) language subtitles that are already defined in the stream itself (or added prior in castMedia once that is in place).

**Comment by benoitdion** [comment]
hey @petrbela @nbennink, any thoughts on merging this PR?

**Comment by petrbela** [comment]
@benoitdion I'll check in a bit. The API is a bit different from what I have in the v4 proposal so I'm thinking where's the best middle ground. I'll probably have more time to look at this later this week.

**Comment by hitaloramon** [comment]
It could have a function of sending an external vtt subtitle, like a subtitleUrl parameter for example.

**Comment by fxdemolisher** [comment]
@hitaloramon there is a way to supply an external subtitle file/url but its outside of the scope of this PR and would have to be a follow up in my opinion.

@petrbela would it be possible to get a review on this PR?

**Comment by fxdemolisher** [comment]
rebased on latest master (v3)

**Pull request merged**